include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_pressure_correction.cpp

```diff
@@ -357,7 +357,7 @@ TimeIntBDFPressureCorrection<dim, Number>::momentum_step()
   /*
    *  explicit variable viscosity update executed prior to calculation of rhs_momentum
    */
-  if(this->param.viscosity_is_variable() and
+  if(this->param.viscous_problem() and this->param.viscosity_is_variable() and
```

Comment:
So far, this code has been executed if `generalized_newtonian_model_data.is_active == true` even if `this->param.viscous_problem() == false` (according to my understanding). Hence, I added another if-statement, `if(this->param.viscous_problem()`. @richardschu could you please double-check this?

Reply:
Yes, that was a bug, thank you!